test/ctia/http/routes/scratchpad_test.clj

```diff
@@ -0,0 +1,287 @@
+          (is (= (:protocol    scratchpad-id)      (:protocol    show-props)))
+          (is (= (:port        scratchpad-id)      (:port        show-props)))
+          (is (= (:path-prefix scratchpad-id) (seq (:path-prefix show-props))))))
+
+      (testing "GET /ctia/scratchpad/:id"
+        (let [response (get (str "ctia/scratchpad/" (:short-id scratchpad-id))
+                            :headers {"Authorization" "45c1f5e3f05d0"})
+              scratchpad (:parsed-body response)]
+          (is (= 200 (:status response)))
+          (is (deep=
+               (assoc new-scratchpad :id (id/long-id scratchpad-id)) scratchpad))))
+
+      (test-query-string-search :scratchpad "description" :description)
+
+      (testing "GET /ctia/scratchpad/external_id/:external_id"
+        (let [response (get (format "ctia/scratchpad/external_id/%s"
+                                    (encode (rand-nth scratchpad-external-ids)))
+                            :headers {"Authorization" "45c1f5e3f05d0"})
+              scratchpads (:parsed-body response)]
+          (is (= 200 (:status response)))
+          (is (deep=
+               [(assoc scratchpad :id (id/long-id scratchpad-id))]
+               scratchpads))))
+
+      (testing "PUT /ctia/scratchpad/:id"
+        (let [with-updates (assoc scratchpad
+                                  :title "modified scratchpad")
+              response (put (str "ctia/scratchpad/" (:short-id scratchpad-id))
+                            :body with-updates
+                            :headers {"Authorization" "45c1f5e3f05d0"})
+              updated-scratchpad (:parsed-body response)]
+          (is (= 200 (:status response)))
+          (is (deep=
+               with-updates
+               updated-scratchpad))))
+
+      (testing "PATCH /ctia/scratchpad/:id"
+        (let [updates {:title "patched scratchpad"}
+              response (patch (str "ctia/scratchpad/" (:short-id scratchpad-id))
+                              :body updates
+                              :headers {"Authorization" "45c1f5e3f05d0"})
+              updated-scratchpad (:parsed-body response)]
+          (is (= 200 (:status response)))
+          (is (= "patched scratchpad"
+                 (:title updated-scratchpad)))))
+
+      ;; -------- partial update operation tests ------------
+
+      ;; observables
+      (testing "POST /ctia/scratchpad/:id/observables :add"
+        (let [new-observables [{:type "ip" :value "42.42.42.42"}]
+              response (post (str "ctia/scratchpad/" (:short-id scratchpad-id) "/observables")
+                             :body {:operation :add
+                                    :observables new-observables}
+                             :headers {"Authorization" "45c1f5e3f05d0"})
+              updated-scratchpad (:parsed-body response)]
+          (is (= 200 (:status response)))
+          (is (subset? (set new-observables)
+                       (set (:observables updated-scratchpad))))))
```

Comment:
I suggest adding a test to check that `new-observables` is not equal to `(:observables updated-scratchpad)`.

Same comment for all `:add` operations on bundle entities and texts.